gradle/setup.gradle

```diff
@@ -112,10 +112,28 @@ publishing {
 			artifact javadocJar
 			//consider adding extra artifacts here, conditionally on submodule's name and perhaps in an afterEvaluate block
 			afterEvaluate {
-				if (project.name == 'reactor-core') {
-					artifact rootProject.tasks.docsZip
+				// Find the docs project (it's not a java project, so we need to lookup it)
+				def docsProject = project.findProject(':docs')
+				if (docsProject) {
+					// Access the docsZip task from the docs project
+					def docsZipTask = docsProject.tasks.findByName('docsZip')
+					if (docsZipTask) {
+						// Add the docsZip task as an artifact
+						artifact docsZipTask
+					}
+				}
+				else {
+					// If the current JDK version is JDK8, the docs project is not loaded (see settings.gradle), so
+					// docsTask is not available. In this case, include the docsZip file path directly as an artifact, if it exists.
+					// (it may exist in case the docs have been previously built using a JDK17 compatible JDK).
+					def docsZipFile = file("${rootDir}/docs/build/distributions/reactor-core-${project.version}-docs.zip")
+					if (docsZipFile.exists()) {
```

Comment:
If it's missing it's still a desired outcome? Should we print a log message whether it is skipped?

Reply:
Added a WARN message, fixed and clarified comments, see ehttps://github.com/reactor/reactor-core/commit/34c854e6ff3b2234a3bb57e858e33b24be719b36